src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,52 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  #s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+                wait_duration = 20e-03  # s
+
+                # Check for presence of rx/t or rz/r keys in pos
+                # In FM mode, the stage will maintain its rotation and tilt, if it is requested to change
+                # it is a different imaging mode
+                # TODO: extend the behavior to other imaging modes
+                if 'r' in pos or 't' in pos:
+                    logging.debug("Position requested to move to includes t (rx/tilt) or r (rz/rotation). "
+                                  "Reported position accuracy is not checked.")
+                    check_pos = False
+
+                # Get the target position in absolute coordinates
+                if not rel:
+                    target_pos = pos
+                else:
+                    target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
+
+                while check_pos:
+                    time.sleep(wait_duration)
+                    if time.time() > expected_end_time:
+                        logging.warning("Stage position after move + %s s is still same as original pos: %s. "
+                                        "Giving up waiting", timeout, orig_pos)
+                        break
```

Comment:
shouldn't this be directly after checking the position? Now you are sleeping for a duration and then break the loop if the time > expected_end_time, however you do not check if the stage position has updated after the sleep

Reply:
good point, I have put this at the end of the while loop so it checks time after it checks stage position